spec/controllers/admin/customers_controller_spec.rb

```diff
@@ -139,99 +139,35 @@ def create_customer(enterprise)
-
-      it "returns with serialized addresses for the customer" do
-        spree_get :addresses, params
-        json_response = JSON.parse(response.body)
-        expect(json_response.keys).to include "bill_address", "ship_address"
-        expect(json_response["bill_address"]["firstname"]).to eq "Dominic"
-        expect(json_response["bill_address"]["address1"]).to eq "123 Lala Street"
-        expect(json_response["ship_address"]["firstname"]).to eq "Dom"
-        expect(json_response["ship_address"]["address1"]).to eq "123 Sesame Street"
-      end
-    end
-
-    context "when I don't manage the customer" do
-      before { params.merge!(customer_id: unmanaged_customer.id) }
-
-      it "redirects to unauthorised" do
-        spree_get :addresses, params
-        expect(response).to redirect_to spree.unauthorized_path
-      end
-    end
-
-    context "when no customer with a matching id exists" do
-      before { params.merge!(customer_id: 1) }
-
-      it "redirects to unauthorised" do
-        spree_get :addresses, params
-        expect(response).to redirect_to spree.unauthorized_path
-      end
-    end
-  end
-
-  describe "#cards" do
-    let(:user) { create(:user) }
-    let!(:enterprise) { create(:enterprise) }
-    let!(:credit_card1) { create(:credit_card, user: user) }
-    let!(:credit_card2) { create(:credit_card) }
-    let(:managed_customer) { create(:customer, enterprise: enterprise) }
-    let(:unmanaged_customer) { create(:customer) }
-    let(:params) { { format: :json } }
+  describe "show" do
+    let(:enterprise) { create(:distributor_enterprise) }
+    let(:another_enterprise) { create(:distributor_enterprise) }
 
-    before { login_as_enterprise_user [enterprise] }
+    context "json" do
+      let!(:customer) { create(:customer, enterprise: enterprise) }
 
-    context "when I manage the customer" do
-      before { params.merge!(id: managed_customer.id) }
+      context "where I manage the customer's enterprise" do
+        render_views
 
-      context "when the customer is not associated with a user" do
-        it "returns with an empty array" do
-          spree_get :cards, params
-          json_response = JSON.parse(response.body)
-          expect(json_response).to eq []
+        before do
+          controller.stub spree_current_user: enterprise.owner
         end
-      end
 
-      context "when the customer is associated with a user" do
-        before { managed_customer.update_attributes(user_id: user.id) }
-
-        it "returns with serialized cards for the customer" do
-          spree_get :cards, params
-          json_response = JSON.parse(response.body)
-          expect(json_response).to be_an Array
-          expect(json_response.length).to be 1
-          expect(json_response.first["id"]).to eq credit_card1.id
+        it "renders the customer as json" do
+          spree_get :show, format: :json, id: customer.id
+          expect(JSON.parse(response.body)["id"]).to eq customer.id
```

Comment:
We have [json_spec](https://github.com/collectiveidea/json_spec) in our Gemfile but we never use it. Do you mind if I remove it?

Reply:
Huh, I didn't even know that was there. I don't mind, I just added a helper method called `json_response` (I think in #2275?), so I am happy to use that, or switch to using `json_spec` if that is a better approach. What do you think?

Reply:
I would stick to `json_response` and get rid of a dependency.